include/jwt-cpp/jwt.h

```diff
@@ -469,24 +469,54 @@ namespace jwt {
 			}
 #endif
 		};
+
+		struct unique_bio_ptr : std::unique_ptr<BIO, decltype(&BIO_free_all)> {
+			unique_bio_ptr() : std::unique_ptr<BIO, decltype(&BIO_free_all)>(BIO_new(BIO_s_mem()), BIO_free_all) {}
+			unique_bio_ptr(const std::string& data)
+				: std::unique_ptr<BIO, decltype(&BIO_free_all)>(
+#if OPENSSL_VERSION_NUMBER <= 0x10100003L
+					  BIO_new_mem_buf(const_cast<char*>(data.data()), static_cast<int>(data.size())), BIO_free_all
+#else
+					  BIO_new_mem_buf(data.data(), static_cast<int>(data.size())), BIO_free_all
+#endif
+				  ) {
+			}
+
+			unique_bio_ptr(const unique_bio_ptr&) = delete;
+			unique_bio_ptr(unique_bio_ptr&&) noexcept = default;
+			~unique_bio_ptr() noexcept = default;
+		};
```

Comment:
I am not really a fan of inheriting `std` classes, especially since in this case they barely do anything and the name does not imply that it creates a mem bio. Maybe something like
```cpp
std::unique_ptr<BIO, decltype(&BIO_free_all)> make_mem_buf_bio(const std::string& data) {
	return std::unique_ptr<BIO, decltype(&BIO_free_all)>(
#if OPENSSL_VERSION_NUMBER <= 0x10100003L
		BIO_new_mem_buf(const_cast<char*>(data.data()), static_cast<int>(data.size())), BIO_free_all
#else
		BIO_new_mem_buf(data.data(), static_cast<int>(data.size())), BIO_free_all
#endif
	);
}
```
Is a better fit ?